### PR TITLE
feat: MultipleDisposable ICollection support + de-duplicate ArgumentExceptionHelper

### PR DESCRIPTION
**Pull request by glennawatson**
This PR bundles two related Primitives cleanups.

## 1. `MultipleDisposable : ICollection<IDisposable>`

Makes `MultipleDisposable` a genuine drop-in for a composite disposable so it can back ReactiveUI's canonical `DisposeWith` pattern (Splat's `DisposeWith` is typed to `ICollection<IDisposable>`), collection-initializer construction, and membership/enumeration queries.

- Adds `Count`, `IsReadOnly`, `Clear()` (disposes contents, stays usable), `Contains`, `CopyTo`, `GetEnumerator` (snapshot-based, thread-safe under the existing gate) and the explicit `IEnumerable.GetEnumerator`. Existing `Add`/`Remove` already satisfy the interface.
- Declares the six new members in `PublicAPI.Shipped.txt` across all eight TFMs.
- Internal empty `new()` constructions become `[]` (the analyzer now prefers a collection expression).
- Full edge-case test coverage (overflow store, `Clear`, `Contains`, `CopyTo`, enumeration, post-dispose) — `MultipleDisposable.cs` line coverage is 99% (only the debugger-only `DebuggerDisplay` is uncovered).

## 2. De-duplicate `ArgumentExceptionHelper`

SonarCloud flagged the two near-identical `ArgumentExceptionHelper` copies (Async + Extensions, ~89 duplicated lines each).

- Replaced with a **single shared polyfill** under `Polyfills/`, compiled **only on net462-net481**.
- On **net8.0+** the `ArgumentExceptionHelper` identifier is **aliased directly to `System.ArgumentNullException`** (via a per-TFM `<Using Alias>`), so the 250 `ThrowIfNull` call sites bind to the BCL method with no shipped wrapper.
- Dropped the dead `ThrowIfNullOrEmpty` (0 uses) and inlined the 3 `Check<T>` uses as `?? throw`.
- Removed the `using ...Internal;` imports the alias orphaned, and 3 no-op `ThrowIfNull(ImmutableArray<>)` guards `CA2264` surfaced once the calls bound to the real BCL method.

## Verification

- Core, all 7 satellite libs, and the test projects build clean across all eight TFMs (net8/9/10/11, net462/472/48/481) with `-warnaserror`.
- Tests pass: core **322**, Async **1215**, Extensions **687** (net9.0; core also on net11.0).

### CODE REVIEW

**Review by ChrisPulman — APPROVED** [review]

File: src/tests/ReactiveUI.Primitives.Tests/DisposableTests.cs [hunk 1]
```diff
@@ -115,4 +114,176 @@ public void MultipleDisposableWithItemsDispose()
...
+
+        disposable.Dispose();
+        Assert.Equal(items.Length, disposedCount);
+    }
+
+    /// <summary>Verifies <see cref="MultipleDisposable.CopyTo"/> validates its arguments.</summary>
+    [Test]
+    public void MultipleDisposableCopyToValidatesArguments()
+    {
+        var disposable = new MultipleDisposable();
```
[the remark this answers is not in the document]
Comment (glennawatson) [reply src/tests/ReactiveUI.Primitives.Tests/DisposableTests.cs:268]: False positive. `MultipleDisposableCopyToValidatesArguments` exists purely to verify `CopyTo`'s **argument** validation: both calls (`CopyTo(null!, 0)` and `CopyTo([], -1)`) throw on argument checks (`ArgumentNullException` / `ArgumentOutOfRangeException`) *before* any element is read, so the container is intentionally empty — its contents are never reached. Pre-populating it would add coverage of nothing and muddy the intent of an argument-validation test. The overflow/enumeration/copy *contents* paths are covered separately by `MultipleDisposableEnumeratesAndCopies` and `MultipleDisposableHandlesOverflow`. Dismissing as a false positive.

File: src/tests/ReactiveUI.Primitives.Tests/DisposableTests.cs [hunk 2]
```diff
@@ -115,4 +114,204 @@ public void MultipleDisposableWithItemsDispose()
...
+        // Clear disposes the remaining items, including the overflow store.
+        disposable.Clear();
+        Assert.Equal(items.Length, disposedCount);
+    }
+
+    /// <summary>Verifies <see cref="MultipleDisposable.CopyTo"/> validates its arguments.</summary>
+    [Test]
+    public void MultipleDisposableCopyToValidatesArguments()
+    {
+        var disposable = new MultipleDisposable();
```
[the remark this answers is not in the document]
Comment (glennawatson) [reply src/tests/ReactiveUI.Primitives.Tests/DisposableTests.cs:268]: False positive. `MultipleDisposableCopyToValidatesArguments` exists solely to verify `CopyTo`'s **argument** validation — `CopyTo(null!, 0)` and `CopyTo([], -1)` both throw on the argument checks (`ArgumentNullException` / `ArgumentOutOfRangeException`) *before* any element is accessed, so the container is intentionally empty and its contents are never reached. Populating it would change nothing the test asserts. Content/overflow copy paths are covered by `MultipleDisposableEnumeratesAndCopies` and `MultipleDisposableHandlesOverflow`. Dismissing.
